### PR TITLE
Styles: Sets Button width and height to inherit from Host

### PR DESCRIPTION
**Pull request by brianchristopherbrady**
This PR makes a simple CSS change to the fluent-button web component. It adds a width of `inherit` to the .control class, allowing custom widths to be applied to the button and enabling the use of properties like justify-self: stretch. This change is especially beneficial when working with media queries for smaller screen widths.

The purpose of this change is to provide more flexibility and control over the button's width, enhancing the responsiveness and adaptability of the component.

### Issue

The problem arises with the visual width of the fluent-button component, specifically related to the .control class within its shadow DOM. As it stands, applying widths directly to the fluent-button component does not affect the width of the actual button displayed.

Currently, in order to adjust the width of the button, it is necessary to access the shadow DOM of the fluent-button component within the fluent-buttons element. This creates a dependency on the internal structure and implementation details of the component, making it less intuitive and potentially error-prone for developers.

This issue highlights the need for a more straightforward and consistent way to modify the width of the fluent-button component, ideally without relying on internal shadow DOM manipulation. By addressing this problem, developers can more easily and consistently control the width of the button, simplifying customization and enhancing the overall usability and maintainability of the component.

### Issue Scenario:
At smaller screen sizes the dialog component adjusts it's layout so that the actions section stacks the buttons in a column rather than a row. The buttons width should also change to fit the size of the container. Without this change applying `justify-self: stretch` to the buttons container does no provide the expected behavior:

image

### Resolution Scenario:
With this change applying `justify-self: stretch` to the buttons container appropriately stretches the width of the buttons to the size of the flex container:

button1

### CODE REVIEW

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/ee984a5df76329a1da80fbe2cb6b20394d26eb7c'>ee984a5df76329a1da80fbe2cb6b20394d26eb7c</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by chrisdholt** [comment]
@brianchristopherbrady you'll need to add a change file here as well.

File: packages/web-components/src/button/button.styles.ts [hunk 1]
```diff
@@ -90,6 +90,7 @@ export const styles = css`
     transition-property: background, border, color;
     transition-timing-function: ${curveEasyEase};
     cursor: pointer;
+    width: inherit;
```
Comment (chrisdholt) [inline packages/web-components/src/button/button.styles.ts:93]: If we're moving inherit here, we should likely set *any* width and height on the host and inherit here. I'd suggest moving the `min-width` to the host as well asa well as min-height.

**Review by procload — APPROVED** [review]

File: change/@fluentui-web-components-290a7b72-4b1b-4b97-81f7-bad3330a0a6b.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-web-components-290a7b72-4b1b-4b97-81f7-bad3330a0a6b.json:1]: ## [🕵 fluentui-web-components-v3 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=28568&commitId=b14fb89f25a181f13bb7eccc2fec7a3ef782c610&env=prod&clientType=FLUENTUI)


  <details><summary><b>Button</b> 18 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Button.Small Icon Only - Dark Mode.pressed.chromium.png|413| Changed|
|Button.Small Icon Only.default.chromium.png|2405| Changed|
|Button.Small With Icon.default.chromium.png|2820| Changed|
|Button.Small With Icon - Dark Mode.hover.chromium.png|946| Changed|
|Button.Small With Icon - Dark Mode.default.chromium.png|887| Changed|
|Button.Small With Icon - RTL.hover.chromium.png|2745| Changed|
|Button.Small With Icon - Dark Mode.pressed.chromium.png|875| Changed|
|Button.Small.hover.chromium.png|2420| Changed|
|Button.Small With Icon.pressed.chromium.png|2590| Changed|
|Button.Small Icon Only - Dark Mode.default.chromium.png|414| Changed|
|Button.Small With Icon - RTL.default.chromium.png|2819| Changed|
|Button.Small.default.chromium.png|2444| Changed|
|Button.Small Icon Only.hover.chromium.png|2395| Changed|
|Button.Small.pressed.chromium.png|2369| Changed|
|Button.Small Icon Only - Dark Mode.hover.chromium.png|422| Changed|
|Button.Small With Icon - RTL.pressed.chromium.png|2590| Changed|
|Button.Small With Icon.hover.chromium.png|2745| Changed|
|Button.Small Icon Only.pressed.chromium.png|2374| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

**Comment by JustSlone** [comment]
@brianchristopherbrady closing as that is the only way to stop notifying reviewers. Please update your PR to point to master following this guide: https://github.com/microsoft/fluentui/pull/31361 and reopen when it's updated. Thanks!

**Pull request closed without merging**